code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -103,6 +91,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	var/icon_off
 	var/type_butt = null
 	var/chem_volume = 15 //Size of a syringe
+	var/genericmes = "USER lights NAME with FLAME"
```

Comment:
wayment, what's the point of this?

Reply:
Cigarettes have different vars to describe different messages when lighting a cigarette with a flame source. This is why lighting something with an igniter has a different visible_message than a zippo, has a different visible_message than a welder, etc.
`if(istype(W, /obj/item/flame/match))
			text = matchmes
		else if(istype(W, /obj/item/flame/lighter/zippo))
			text = zippomes
		else if(istype(W, /obj/item/flame/lighter))
			text = lightermes
		else if(W.iswelder())
			text = weldermes
		else if(istype(W, /obj/item/device/assembly/igniter))
			text = ignitermes`

Genericmes was added as an else as a catch-all to anything that didn't fall under these typepaths so a visible_message would be displayed.

Reply:
I guess matchmes could be used as the generic but this way we could give matches unique messages in the future.